packages/react-composites/tests/browser/call/app/mocks/MockCallAdapter.ts

```diff
@@ -0,0 +1,248 @@
+  off(event: string): void {
+    return;
+  }
+}
+
+/**
+ * Helper function to convert array of TestRemoteParticipant to record of RemoteParticipantState
+ * @param mockRemoteParticipants - array of TestRemoteParticipant
+ * @returns Record of RemoteParticipantState
+ */
+const convertTestParticipantsToCallAdapterStateParticipants = (
+  testRemoteParticipants: TestRemoteParticipant[]
+): Record<string, RemoteParticipantState> => {
+  const remoteParticipants: Record<string, RemoteParticipantState> = {};
+
+  testRemoteParticipants.forEach((testRemoteParticipant, i) => {
+    remoteParticipants[i] = {
+      identifier: { kind: 'communicationUser', communicationUserId: `${i}` },
+      state: 'Connected',
+      videoStreams: {
+        1: {
+          id: 1,
+          mediaStreamType: 'Video',
+          isAvailable: !!testRemoteParticipant.isVideoStreamAvailable
+        }
+      },
+      isMuted: !!testRemoteParticipant.isMuted,
+      isSpeaking: !!testRemoteParticipant.isSpeaking,
+      displayName: testRemoteParticipant.displayName
+    };
+  });
+
+  return remoteParticipants;
+};
+
+/**
+ * Helper function to add mock video element to RemoteParticipantState
+ * @param remoteParticipant - RemoteParticipantState
+ * @returns void
+ */
+const addMockVideo = (remoteParticipant: RemoteParticipantState): void => {
+  for (const videoStream of Object.values(remoteParticipant.videoStreams)) {
+    if (videoStream.isAvailable) {
+      // The video is an empty html5 video element with a colored background. To ensure a consistent color the background color
+      // is the hex representation of the participant's display name.
+      const mockVideoElement = document.createElement('div');
+      mockVideoElement.innerHTML = '<span />';
+      mockVideoElement.style.width = decodeURIComponent('100%25');
+      mockVideoElement.style.height = decodeURIComponent('100%25');
+      mockVideoElement.style.background = stringToHexColor(remoteParticipant.displayName);
+      mockVideoElement.style.backgroundPosition = 'center';
+      mockVideoElement.style.backgroundRepeat = 'no-repeat';
+      const view: VideoStreamRendererViewState = { scalingMode: 'Crop', isMirrored: false, target: mockVideoElement };
+      videoStream.view = view;
+    }
+  }
+};
+/**
+ * Helper function to randomly choose a background color for mocking a video stream
+ * @param str - input string
+ * @returns hex color code as a string
+ */
+const stringToHexColor = (str: string): string => {
+  let hash = 0;
+  for (let i = 0; i < str.length; i++) {
+    hash = str.charCodeAt(i) + ((hash << 5) - hash);
+  }
+  let colour = '#';
+  for (let i = 0; i < 3; i++) {
+    const value = (hash >> (i * 8)) & 0xff;
+    colour += ('00' + value.toString(16)).substr(-2);
+  }
+  return colour;
+};
+
+/**
+ * Default call adapter state that the {@link MockCallAdapter} class is initialized with
+ */
+const defaultCallAdapterState: CallAdapterState = {
```

Comment:
I think that this setup is quite hidden / hard to follow:
- test creates a `TestCallingState`
- passes it in as qargs to app
- app creates a new MockCallAdapter
- which then populates a whole bunch of values in the adapter state from this default.

Instead, consider the following flow:
- test uses a helper method that creates a `CallAdapterState`. The helper could simply accept a list of remote participants, filling in all the rest with defaults like here: `createMockStateWithRemoteParticipants(...)`.
- app accepts a `CallAdapterState` via quargs, and directly uses that.

This flow is less magical for somebody reading the test.

Reply:
Yes, I agree that the flow is a bit convoluted. But I was unable to figure out how to import `CallAdapterState` into 
'packages/react-composites/tests/browser/call/app' without causing the tests to fail with vague error 'window is not defined' when I run the tests. I will create a bug to simplify the setup. But for now I think this should suffice as the tests are working.